# Video Game Analysis

As a data analyst at the video game company TerpStudios, the purpose of this notebook is to make data-driven recommendations for TerpStudios on what they should do for the release of their next video game. In order to do so, we will use a data set that provides information about the best selling video games and their sales across global markets. The first step is to cleanse and prepare the data to rid any missing information or potential duplicates. After, we will use the cleaned data set for analysis to find information about video game sales and use correlation matrices to understand the relationships between sales across different markets. Finally, we will make our data-driven recommendation for the company based on the information gathered in the data analysis.

In order to clean and analyze the data using dataframes, import the NumPy and Pandas libraries to load the csv data for prepartion and further analysis. Using the NumPy library, import the missing values as NA to use for data that would be considered null.

In [20]:
import pandas as pd
import numpy as np
from numpy import nan as NA

## Initial Data Load

Get a brief sense of the data set by viewing the first few observations in the dataframe.

In [21]:
# Load data to index the rankings of video games and visualize the output for further analysis
game_df = pd.read_csv('HW4-video-game-sales.csv')
game_df.index = game_df['Rank']
# Delete as the rank is now repetitive
del game_df['Rank']
game_df.head(30)

Name  \
Rank                                                 
1                                       Wii Sports   
2                                              NaN   
2                                Super Mario Bros.   
3                                   Mario Kart Wii   
4                                Wii Sports Resort   
5                         Pokemon Red/Pokemon Blue   
6                                           Tetris   
7                            New Super Mario Bros.   
8                                          Missing   
8                                         Wii Play   
9                        New Super Mario Bros. Wii   
10                                       Duck Hunt   
11                                      Nintendogs   
12                                   Mario Kart DS   
13                     Pokemon Gold/Pokemon Silver   
14                                         Wii Fit   
15                                    Wii Fit Plus   
16                                             NaN   
17                              Grand Theft Auto V   
18                   Grand Theft Auto: San Andreas   
19                               Super Mario World   
20    Brain Age: Train Your Brain in Minutes a Day   
21                   Pokemon Diamond/Pokemon Pearl   
22                                Super Mario Land   
22                                             NaN   
23                             Super Mario Bros. 3   
24                              Grand Theft Auto V   
25                     Grand Theft Auto: Vice City   
26                   Pokemon Ruby/Pokemon Sapphire   
27                     Pokemon Black/Pokemon White   

               Publisher:Platform     Year         Genre NA_Sales EU_Sales  \
Rank                                                                         
1                    Nintendo:Wii     2006        Sports    41.49    29.02   
2                             NaN      NaN           NaN      NaN      NaN   
2                    Nintendo:NES     1985      Platform    29.08     3.58   
3                    Nintendo:Wii     2008        Racing    15.85    12.88   
4                    Nintendo:Wii     2009        Sports    15.75    11.01   
5                     Nintendo:GB     1996  Role-Playing    11.27     8.89   
6                     Nintendo:GB     1989        Puzzle     23.2     2.26   
7                     Nintendo:DS     2400      Platform    11.38     9.23   
8                         Missing  Missing       Missing  Missing  Missing   
8                    Nintendo:Wii     2006          Misc    14.03      9.2   
9                    Nintendo:Wii     2009      Platform    14.59     7.06   
10                   Nintendo:NES     1984       Shooter    26.93     0.63   
11                    Nintendo:DS    12005    Simulation     9.07       11   
12                    Nintendo:DS     2005        Racing     9.81     7.57   
13                    Nintendo:GB     1999  Role-Playing      NaN      NaN   
14                   Nintendo:Wii     2007        Sports     8.94     8.03   
15                   Nintendo:Wii     2009        Sports     9.09     8.59   
16    Microsoft Game Studios:X360     2010          Misc    14.97     4.94   
17       Take-Two Interactive:PS3     2013        Action     7.01     9.27   
18       Take-Two Interactive:PS2     2004        Action     9.43      0.4   
19                  Nintendo:SNES     1990      Platform    12.78     3.75   
20                    Nintendo:DS     2005          Misc     4.75     9.26   
21                    Nintendo:DS     2006  Role-Playing     6.42     4.52   
22                    Nintendo:GB     1989      Platform    10.83     2.71   
22                            NaN      NaN           NaN      NaN      NaN   
23                   Nintendo:NES     1988      Platform     9.54     3.44   
24      Take-Two Interactive:X360     2013        Action     9.63     5.31   
25       Take-Two Interactive:PS2     2002        Actio

## Clean Data Observations

We will start by finding the percentage of observations in the data set that is clean. In order to do so, we must find which ranks are duplicated as well as what data is missing or invalid or does not fit into the context of the variable. Using the findings we will describe the processes used to make the determination.

In [22]:
# Use dataframe with only the first one undred  observations to scan for dirty data
hundred_observations = game_df.copy().head(100)
hundred_observations

Name              Publisher:Platform  \
Rank                                                                           
1                                 Wii Sports                    Nintendo:Wii   
2                                        NaN                             NaN   
2                          Super Mario Bros.                    Nintendo:NES   
3                             Mario Kart Wii                    Nintendo:Wii   
4                          Wii Sports Resort                    Nintendo:Wii   
...                                      ...                             ...   
92                      Super Mario Galaxy 2                    Nintendo:Wii   
93              Star Wars Battlefront (2015)             Electronic Arts:PS4   
94            Call of Duty: Advanced Warfare                  Activision:PS4   
95      The Legend of Zelda: Ocarina of Time                    Nintendo:N64   
96    Crash Bandicoot 2: Cortex Strikes Back  Sony Computer Entertainment:PS   

      Year     Genre NA_Sales EU_Sales JP_Sales Other_Sales  
Rank                                                         
1     2006    Sports    41.49    29.02     3.77        8.46  
2      NaN       NaN      NaN      NaN      NaN         NaN  
2     1985  Platform    29.08     3.58     6.81        0.77  
3     2008    Racing    15.85    12.88     3.79        3.31  
4     2009    Sports    15.75    11.01     3.28        2.96  
...    ...       ...      ...      ...      ...         ...  
92    2010  Platform     3.66     2.42     0.98        0.64  
93    2015   Shooter     2.93     3.29     0.22        1.23  
94    2014   Shooter      2.8      3.3     0.14        1.37  
95    1998    Action      4.1     1.89     1.45        0.16  
96    1997  Platform     3.78     2.17     1.31        0.31  

[100 rows x 8 columns]

From viewing the data from afar we can see that at least one observation contains the string 'Missing'. We will count this as dirty data by replacing the value with NA. We will consider any observation with NA to be dirty. Subsequently, any values that do not have valid data points will be deleted. We will also make sure to drop any duplicates if there are any. 

In [23]:
# Other values that are not strictly NA must also be removed to cleanse the data
hundred_observations.replace(['N/A','Missing'], NA, inplace=True)
# Drop any observations that do not have a sufficient enough number of valid data points
hundred_observations.dropna(inplace=True)
hundred_observations.drop_duplicates()
hundred_observations.head(60)

Name  \
Rank                                                 
1                                       Wii Sports   
2                                Super Mario Bros.   
3                                   Mario Kart Wii   
4                                Wii Sports Resort   
5                         Pokemon Red/Pokemon Blue   
6                                           Tetris   
7                            New Super Mario Bros.   
8                                         Wii Play   
9                        New Super Mario Bros. Wii   
11                                      Nintendogs   
12                                   Mario Kart DS   
14                                         Wii Fit   
15                                    Wii Fit Plus   
17                              Grand Theft Auto V   
18                   Grand Theft Auto: San Andreas   
19                               Super Mario World   
20    Brain Age: Train Your Brain in Minutes a Day   
21                   Pokemon Diamond/Pokemon Pearl   
22                                Super Mario Land   
23                             Super Mario Bros. 3   
24                              Grand Theft Auto V   
25                     Grand Theft Auto: Vice City   
26                   Pokemon Ruby/Pokemon Sapphire   
27                     Pokemon Black/Pokemon White   
28     Brain Age 2: More Training in Minutes a Day   
29                          Gran Turismo 3: A-Spec   
30                  Call of Duty: Modern Warfare 3   
31         Pokémon Yellow: Special Pikachu Edition   
32                         Call of Duty: Black Ops   
33                             Pokemon X/Pokemon Y   
34                       Call of Duty: Black Ops 3   
35                      Call of Duty: Black Ops II   
36                      Call of Duty: Black Ops II   
37                  Call of Duty: Modern Warfare 2   
38                  Call of Duty: Modern Warfare 3   
39                            Grand Theft Auto III   
40                         Super Smash Bros. Brawl   
41                         Call of Duty: Black Ops   
42                     Animal Crossing: Wild World   
43                                    Mario Kart 7   
44                                          Halo 3   
45                              Grand Theft Auto V   
46            Pokemon HeartGold/Pokemon SoulSilver   
47                                  Super Mario 64   
48                                  Gran Turismo 4   
49                              Super Mario Galaxy   
50       Pokemon Omega Ruby/Pokemon Alpha Sapphire   
51              Super Mario Land 2: 6 Golden Coins   
52                             Grand Theft Auto IV   
53                                    Gran Turismo   
54                             Super Mario 3D Land   
55                                  Gran Turismo 5   
56                  Call of Duty: Modern Warfare 2   
57                             Grand Theft Auto IV   
58                           Super Mario All-Stars   
59               Pokemon FireRed/Pokemon LeafGreen   
60                                  Super Mario 64   
61                                    Just Dance 3   
62                            Call of Duty: Ghosts   
63                                     Halo: Reach   

                   Publisher:Platform   Year         Genre NA_Sales EU_Sales  \
Rank                                                                           
1                        Nintendo:Wii   2006        Sports    41.49    29.02   
2                        Nintendo:NES   1985      Platform    29.08     3.58   
3                        Nintendo:Wii   2008        Racing    15.85    12.88   
4                        Nintendo:Wii   2009        Sports    15.75    11.01   
5                         Nintendo:GB   1996  Role-Playing    11.27     8.89   
6                         Nintendo:GB   1989        Puzzle     23.2     2.26   
7                         Nintendo:DS   2400      Platfor

Some of the year values also seem to be too large. We will check each value in the year column to see which year values seem off. 

In [24]:
# Determine the amount of faulty year data
wrong_year_check = len(hundred_observations[(hundred_observations['Year'].astype(float) < 1970) | (hundred_observations['Year'].astype(float) > 2022)])
print(f'{wrong_year_check} observations have years that are either outside of the time when video games were or have yet to be sold.')

2 observations have years that are either outside of the time when video games were or have yet to be sold.


In [25]:
# Find how much of the data needs to be cleaned
first_hundred_observations = 100
print(f'{(len(hundred_observations) - wrong_year_check)/first_hundred_observations*100:.2f}% of the first one hundred observations contain clean data.')

91.00% of the first one hundred observations contain clean data.


### Analysis:

The process used to find the percentage of clean data includes finding the amounts of NA or missing data, finding the amount of duplicated data, and finding which data was abnormal or unexpected, depending on which column the data was in. Determining the sum of all values that are NA would be considered dirty data. Finding the sum of all duplicated indexes indicates would indicate data that is repetitive and not needed. The sum of all columns that contain an NA value would also be considered uncleaned as there is missing data that would need to be imputed to provide all variables for each observation. Abnormal data points would also be considered as untidied which can be found by viewing the first few observations of the data to see if any of the years do not have four digits or are years that are not representative of the actual data. As concluded above, 91% of the first one hundred observations contain clean data.

## Data Cleansing and Preparation

In order to prepare the video game data for analysis, the data must first be cleansed and tidied by removing any missing data. Missing data can come in several different forms which means that any null substitute values must be replaced with NA. To prepare the data for in-depth analysis adding the total sales numbers across all regions will be helpful and must be created as an additional column in the data set. Using float values instead of string values will be important when sorting and comparing the data numerically.

Delete observations that do not have at least seven values in order to make proper use of data. Missing seven or more data points nullifies the value of having the observation in the data set because there is so little information. Make sure that the inplace argument is used in order to make permanent changes to the dataframe.

In [26]:
# Drop missing values to only utilize observations that would be important for data analysis
game_df.replace(['N/A','Missing'], NA, inplace=True)
game_df.dropna(thresh=7, inplace=True)
game_df.head(30)

Name  \
Rank                                                 
1                                       Wii Sports   
2                                Super Mario Bros.   
3                                   Mario Kart Wii   
4                                Wii Sports Resort   
5                         Pokemon Red/Pokemon Blue   
6                                           Tetris   
7                            New Super Mario Bros.   
8                                         Wii Play   
9                        New Super Mario Bros. Wii   
10                                       Duck Hunt   
11                                      Nintendogs   
12                                   Mario Kart DS   
14                                         Wii Fit   
15                                    Wii Fit Plus   
16                                             NaN   
17                              Grand Theft Auto V   
18                   Grand Theft Auto: San Andreas   
19                               Super Mario World   
20    Brain Age: Train Your Brain in Minutes a Day   
21                   Pokemon Diamond/Pokemon Pearl   
22                                Super Mario Land   
23                             Super Mario Bros. 3   
24                              Grand Theft Auto V   
25                     Grand Theft Auto: Vice City   
26                   Pokemon Ruby/Pokemon Sapphire   
27                     Pokemon Black/Pokemon White   
28     Brain Age 2: More Training in Minutes a Day   
29                          Gran Turismo 3: A-Spec   
30                  Call of Duty: Modern Warfare 3   
31         Pokémon Yellow: Special Pikachu Edition   

                   Publisher:Platform   Year         Genre NA_Sales EU_Sales  \
Rank                                                                           
1                        Nintendo:Wii   2006        Sports    41.49    29.02   
2                        Nintendo:NES   1985      Platform    29.08     3.58   
3                        Nintendo:Wii   2008        Racing    15.85    12.88   
4                        Nintendo:Wii   2009        Sports    15.75    11.01   
5                         Nintendo:GB   1996  Role-Playing    11.27     8.89   
6                         Nintendo:GB   1989        Puzzle     23.2     2.26   
7                         Nintendo:DS   2400      Platform    11.38     9.23   
8                        Nintendo:Wii   2006          Misc    14.03      9.2   
9                        Nintendo:Wii   2009      Platform    14.59     7.06   
10                       Nintendo:NES   1984       Shooter    26.93     0.63   
11                        Nintendo:DS  12005    Simulation     9.07       11   
12                        Nintendo:DS   2005        Racing     9.81     7.57   
14                       Nintendo:Wii   2007        Sports     8.94     8.03   
15                       Nintendo:Wii   2009        Sports     9.09     8.59   
16        Microsoft Game Studios:X360   2010          Misc    14.97     4.94   
17           Take-Two Interactive:PS3   2013        Action     7.01     9.27   
18           Take-Two Interactive:PS2   2004        Action     9.43      0.4   
19                      Nintendo:SNES   1990      Platform    12.78     3.75   
20                        Nintendo:DS   2005          Misc     4.75     9.26   
21                        Nintendo:DS   2006  Role-Playing     6.42     4.52   
22                        Nintendo:GB   1989      Platform    10.83     2.71   
23                       Nintendo:NES   1988      Platform     9.54     3.44   
24          Take-Two Interactive:X360   2013        Action     9.63     5.31   
25           Take-Two Interactive:PS2   2002        Action     8.41     5.49   
26                       Nintendo:GBA   2002  Role-Playing     6.06      3.9   
27                        Nintendo:DS   2010  Role-Playing     5.57     3.28   
28                        Nintendo:DS   2005        Puzzle     3.44     5.36 

Each of the sales are provided in millions per the four listed regions which include North America, Europe, Japan, and all other markets around the globe. In order to find the total sales number, the four sales columns must be added up to provide the total sales for each video game. Convert the dtypes for sales to floats in order to add each sales market together.

In [27]:
# Create total global sales column to visualize the sum of all sales locations for analysis regarding the percentage of the total market
game_df['Global_Sales'] = game_df.loc[:,'NA_Sales': 'Other_Sales'].astype(float).sum(axis=1)
game_df

Name Publisher:Platform  \
Rank                                                                         
1                                            Wii Sports       Nintendo:Wii   
2                                     Super Mario Bros.       Nintendo:NES   
3                                        Mario Kart Wii       Nintendo:Wii   
4                                     Wii Sports Resort       Nintendo:Wii   
5                              Pokemon Red/Pokemon Blue        Nintendo:GB   
...                                                 ...                ...   
16594                Woody Woodpecker in Crazy Castle 5          Kemco:GBA   
16595                     Men in Black II: Alien Escape      Infogrames:GC   
16596  SCORE International Baja 1000: The Official Game     Activision:PS2   
16597                                        Know How 2        7G//AMES:DS   
16598                                  Spirits & Spells        Wanadoo:GBA   

       Year         Genre NA_Sales EU_Sales JP_Sales Other_Sales  Global_Sales  
Rank                                                                            
1      2006        Sports    41.49    29.02     3.77        8.46         82.74  
2      1985      Platform    29.08     3.58     6.81        0.77         40.24  
3      2008        Racing    15.85    12.88     3.79        3.31         35.83  
4      2009        Sports    15.75    11.01     3.28        2.96         33.00  
5      1996  Role-Playing    11.27     8.89    10.22           1         31.38  
...     ...           ...      ...      ...      ...         ...           ...  
16594  2002      Platform     0.01        0        0           0          0.01  
16595  2003       Shooter     0.01        0        0           0          0.01  
16596  2008        Racing        0        0        0           0          0.00  
16597  2010        Puzzle        0     0.01        0           0          0.01  
16598  2003      Platform     0.01        0        0           0          0.01  

[16597 rows x 9 columns]

In order to get a sense of the best selling games worldwide, we will sort the dataframe by global sales while including the rank index and the video game title, publisher, platform, and global sales columns.

In [28]:
# Find which video games sold the most worldwide
game_df[['Name', 'Publisher:Platform', 'Global_Sales']].sort_values('Global_Sales', ascending=False).head(15)

Name           Publisher:Platform  Global_Sales
Rank                                                                      
1                    Wii Sports                 Nintendo:Wii         82.74
2             Super Mario Bros.                 Nintendo:NES         40.24
3                Mario Kart Wii                 Nintendo:Wii         35.83
4             Wii Sports Resort                 Nintendo:Wii         33.00
5      Pokemon Red/Pokemon Blue                  Nintendo:GB         31.38
6                        Tetris                  Nintendo:GB         30.26
7         New Super Mario Bros.                  Nintendo:DS         30.01
8                      Wii Play                 Nintendo:Wii         29.01
9     New Super Mario Bros. Wii                 Nintendo:Wii         28.61
10                    Duck Hunt                 Nintendo:NES         27.84
11                   Nintendogs                  Nintendo:DS         24.75
12                Mario Kart DS                  Nintendo:DS         23.43
14                      Wii Fit                 Nintendo:Wii         22.72
15                 Wii Fit Plus                 Nintendo:Wii         22.00
16                          NaN  Microsoft Game Studios:X360         21.82

In the event that we would like to sort the data by publisher, the analysis process would be quite difficult as the publisher is also attatched to the platform. In order to alleviate any future issues, we will break apart the variables into two distinct columns to improve the organization of the data.

In [29]:
# Split column to ease the analysis process by being able to sort through distinct publishers and platforms
game_df['Publisher'] = game_df['Publisher:Platform'].str.split(':').str[0]
game_df['Platform'] = game_df['Publisher:Platform'].str.split(':').str[1]
# Delete as the column is now repetitve
del game_df['Publisher:Platform']
game_df

Name  Year         Genre  \
Rank                                                                          
1                                            Wii Sports  2006        Sports   
2                                     Super Mario Bros.  1985      Platform   
3                                        Mario Kart Wii  2008        Racing   
4                                     Wii Sports Resort  2009        Sports   
5                              Pokemon Red/Pokemon Blue  1996  Role-Playing   
...                                                 ...   ...           ...   
16594                Woody Woodpecker in Crazy Castle 5  2002      Platform   
16595                     Men in Black II: Alien Escape  2003       Shooter   
16596  SCORE International Baja 1000: The Official Game  2008        Racing   
16597                                        Know How 2  2010        Puzzle   
16598                                  Spirits & Spells  2003      Platform   

      NA_Sales EU_Sales JP_Sales Other_Sales  Global_Sales   Publisher  \
Rank                                                                     
1        41.49    29.02     3.77        8.46         82.74    Nintendo   
2        29.08     3.58     6.81        0.77         40.24    Nintendo   
3        15.85    12.88     3.79        3.31         35.83    Nintendo   
4        15.75    11.01     3.28        2.96         33.00    Nintendo   
5        11.27     8.89    10.22           1         31.38    Nintendo   
...        ...      ...      ...         ...           ...         ...   
16594     0.01        0        0           0          0.01       Kemco   
16595     0.01        0        0           0          0.01  Infogrames   
16596        0        0        0           0          0.00  Activision   
16597        0     0.01        0           0          0.01    7G//AMES   
16598     0.01        0        0           0          0.01     Wanadoo   

      Platform  
Rank            
1          Wii  
2          NES  
3          Wii  
4          Wii  
5           GB  
...        ...  
16594      GBA  
16595       GC  
16596      PS2  
16597       DS  
16598      GBA  

[16597 rows x 10 columns]

## Data Analysis

Now that the data set has been cleaned and prepared, we will make insights into the data by making findings to evaluate the data.

### Top Video Games Published By Nintendo

In [30]:
# Find how many of the top video games were published by Nintendo to get a sense of the data
nintendo_games = game_df['Publisher'].value_counts()['Nintendo']
print(f'The amount of top selling games that are published by Nintendo is {nintendo_games}')

The amount of top selling games that are published by Nintendo is 702


### Top Video Games With The Word <i>Super</i> In The Title

In [31]:
# Determine how many games have super in the title to evaluate the data
games_with_super = game_df['Name'].str.contains('Super').sum()
print(f'{games_with_super} video games have the word "Super" in the title')

385 video games have the word "Super" in the title


### Video Games That Have Reached Sales Of At Least $20,000,000 In At Least One Market Alone

In [32]:
# Convert sales values to float in order to compare sales numerically
game_df[['NA_Sales', 'EU_Sales', 'JP_Sales', 'Other_Sales']] = game_df[['NA_Sales', 'EU_Sales', 'JP_Sales', 'Other_Sales']].astype(float)
# Find which markets sold twenty million dollars alone to determine which types of games were extremely popular in specific regions
games_twenty_mil = len(game_df[(game_df['NA_Sales'] >= 20) | (game_df['EU_Sales'] >= 20) | (game_df['JP_Sales'] >= 20) | (game_df['Other_Sales'] >= 20)])
print(f'{games_twenty_mil} video games have reached sales of at least twenty million dollars in at least one market')

4 video games have reached sales of at least twenty million dollars in at least one market


### Percentage Of Video Games Released In 2002 And Beyond That Have Made Over $2,000,000

In [39]:
# Convert year values to float in order to compare years numerically
game_df['Year'] = game_df['Year'].astype(float)
# Determine the percentage of games that made at least two million dollars in 2002 and beyond in order to get a sense of how well modern games sell
percent_two_mil = len(game_df[(game_df['Year'] >= 2002) & (game_df['Global_Sales'] >= 2)])/len(game_df[game_df['Year'] >= 2002])
print(f'For games that were created in the year 2002 and beyond, the percentage of games that made over two million dollars in global sales is {percent_two_mil*100:.2f}%')

For games that were created in the year 2002 and beyond, the percentage of games that made over two million dollars in global sales is 4.50%


### Sales By Market Correlation Matrix Across All Years

In [34]:
sales_categories = game_df[['NA_Sales', 'EU_Sales', 'JP_Sales', 'Other_Sales', 'Global_Sales']]
sales_categories.corr()

NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales
NA_Sales      1.000000  0.765962  0.443597     0.652019      0.940619
EU_Sales      0.765962  1.000000  0.427718     0.727259      0.902274
JP_Sales      0.443597  0.427718  1.000000     0.290052      0.605417
Other_Sales   0.652019  0.727259  0.290052     1.000000      0.755407
Global_Sales  0.940619  0.902274  0.605417     0.755407      1.000000

#### Analysis:

The first observation gathered from the above correlation matrix is that global sales are highly correlated with North American sales at a correlation coefficient of 0.940619 which is very close to 1, indicating a positive linear correlation between the two variables. The inference that can be made about said observation is that a very high percentage of global sales of video games are located in North America due to the wealth of the countries that the region is comprised of as well as the large population of people. Since more people are making higher incomes on average, more people can afford additional luxuries like video games which may explain why the correlation is so high with global sales.

The second observation gathered from the above correlation matrix is that regional Japanese sales are nowhere near as correlated with global sales as the rest of the provided regions. With a correlation coefficient of 0.605417 between Japanese sales and global sales, there is much less correlation between the two variables than any of the other regions. The inference that can be made about said observation is that the Japanese region is comprised of only one country, Japan, which has a significantly lower population than any of the other regions. As a result, sales will not make up a great portion of total sales as there is simply not enough consumers to generate as much in sales on a global level.

### Sales By Market Correlation Matrix In 2000

In [35]:
sales_categories[game_df['Year'] == 2000].corr()

NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales
NA_Sales      1.000000  0.866555  0.139115     0.691317      0.900914
EU_Sales      0.866555  1.000000  0.108401     0.748552      0.866584
JP_Sales      0.139115  0.108401  1.000000     0.226852      0.512328
Other_Sales   0.691317  0.748552  0.226852     1.000000      0.762164
Global_Sales  0.900914  0.866584  0.512328     0.762164      1.000000

### Sales By Market Correlation Matrix In 2015

In [36]:
sales_categories[game_df['Year'] == 2015].corr()

NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales
NA_Sales      1.000000  0.736844  0.123217     0.866672      0.911533
EU_Sales      0.736844  1.000000  0.132160     0.926506      0.922360
JP_Sales      0.123217  0.132160  1.000000     0.126803      0.307129
Other_Sales   0.866672  0.926506  0.126803     1.000000      0.956571
Global_Sales  0.911533  0.922360  0.307129     0.956571      1.000000

#### Analysis:

The first observation gathered from the two above correlation matrices is that the correlation between Japanese sales and total global sales has decreased significantly from a correlation coefficient of 0.512328 in 2000 to a correlation coefficient of 0.307129 in 2015. Over time the correlation between the two variables has decreased significantly. The inference that can be made about said observation is that because of the fact that Japanese population growth rate has not been able to keep up with the population growth rates around the world, a smaller portion of total video game consumers are located in Japan as young people who buy video games are a much slimmer percent of the population.

The second observation gathered from the two above correlation matrices is that the regions outside of North America, Europe, and Japan have overtaken North American Sales as the most correlated variable between them and total global sales going from a correlation coefficient of 0.762164 in 2000 to a correlation coefficient of 0.956571 in 2015. The inference that can be made about said observation is that the immense population growth and economic development of developing countries have created a greater market as more people are able to afford video games in these regions. As a whole, the world population is continuing to grow except for countries in established nations like those in the west and Japan which explains why other sales outside of these regions are so highly correlated with the total video game sales.

### Data-Driven Recommendation For Next Video Game

For the next video game that TerpStudios plans on making, the company should focus on releasing a game that appeals to markets other than North America, Europe, and Japan. The reason, as mentioned briefly above, as to why TerpStudios should focus on attracting consumers in developing regions is because of the population and economic development in these countries. The correlation coefficient between the 'Other_Sales' variable and the 'Global_Sales' variable for games released in 2015 was 0.956571 and is continuing to grow in parallel with the growth in these regions. 

In addition to where TerpStudios should target their next game, the genre variable is also important in finding which types of genres sell best. The reason for this conclusion is based first on one of the most successful publishers, Nintendo, and their most popular genre of games, platformers. Using the data gathered from a large publisher like Nintendo gives a good basis on what genres have the potential to sell very well. As discovered above, 702 of the top 16,597 titles were Nintendo games, proving that Nintendo is a reliable company to base the company's methods on. To perform some further analysis into what genre made Nintendo the most in sales, the data frame below shows the grouped genres ordered by the global sales figures which shows that platformers made the most amount of global sales for Nintendo which is comprised of 15.95% of Nintendo total sales.

In [37]:
game_df[game_df['Publisher'] == 'Nintendo'].groupby('Genre').sum().sort_values('Global_Sales', ascending=False)

Year  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
Genre                                                               
Platform      222540.0    220.14     85.21    102.36        19.47   
Role-Playing  208721.0     96.63     57.74     95.04        12.36   
Sports        110069.0     98.77     66.18     35.87        17.18   
Misc          200625.0     61.98     51.62     55.25        11.78   
Racing         73967.0     73.55     39.75     29.22         8.81   
Action        156516.0     63.49     29.02     29.24         6.48   
Puzzle        148134.0     55.74     26.42     37.09         5.53   
Simulation     66156.0     29.70     26.05     23.67         5.86   
Shooter        48030.0     51.39      9.85      6.07         1.92   
Fighting       36031.0     27.10      8.64     14.94         2.65   
Adventure      70161.0     17.72      7.66      9.01         1.28   
Strategy       62111.0     11.66      4.42     10.46         0.83   

              Global_Sales  
Genre                       
Platform            427.18  
Role-Playing        261.77  
Sports              218.00  
Misc                180.63  
Racing              151.33  
Action              128.23  
Puzzle              124.78  
Simulation           85.28  
Shooter              69.23  
Fighting             53.33  
Adventure            35.67  
Strategy             27.37

In [38]:
print(f'{len(game_df[(game_df["Publisher"] == "Nintendo") & (game_df["Genre"] == "Platform")])/len(game_df[game_df["Publisher"] == "Nintendo"])*100:.2f}% of Nintendo games are platformers')

15.95% of Nintendo games are platformers
